src/utils/SpokeUtils.ts

```diff
@@ -177,3 +178,45 @@ export function relayFilledAmount(
 
   return spokePool.relayFills(hash, { blockTag });
 }
+
+export async function findFillBlock(
+  spokePool: Contract,
+  relayData: RelayData,
+  lowBlockNumber: number,
+  highBlockNumber?: number
+): Promise<number | undefined> {
+  const { provider } = spokePool;
+  highBlockNumber ??= await provider.getBlockNumber();
```

Comment:
todo: Should probably assert `highBlockNumber > lowBlockNumber`.

Reply:
https://github.com/across-protocol/sdk-v2/pull/488/commits/1e6d86d95b2e1377959bb7100bbdb851c8923594